lib/main.js

```diff
@@ -39,6 +39,33 @@ let gSuperfishPEM =
 // Otherwise, the distrust bit will be lost.
 let gSuperfishCert = null;
 
+// We need to ensure the proxy is uninstalled or we DoS users
+function shouldAttemptDistrust() {
+  try {
+    // Check we're on Windows
+    var osString = Cc["@mozilla.org/xre/app-info;1"]
+                     .getService(Ci.nsIXULRuntime).OS;
+    if ('WINNT' !== osString) {
+      log("Superfish hotfix is windows only");
+      return false;
+    }
+
+    // Test for the superfish uninstall key; if present, it's still
+    // installed
+    let registry = Cc["@mozilla.org/windows-registry-key;1"]
+                     .createInstance(Ci.nsIWindowsRegKey);
+    registry.open(Ci.nsIWindowsRegKey.ROOT_KEY_LOCAL_MACHINE,
+                  "SOFTWARE\\MICROSOFT\\WINDOWS\\CURRENTVERSION\\UNINSTALL",
+                  Ci.nsIWindowsRegKey.ACCESS_READ);
+    if (registry.hasChild("Superfish Inc. VisualDiscovery")) {
+      return false;
+    }
+  } catch (ex) {
+
```

Comment:
It would be nice to log the exception here (or, not try/catch here at all, since shouldAttemptDistrust is already called in a try/catch block).